### DISCUSSION TITLE
Keycloak 25 refuses to export users with no further explanation

### DISCUSSION THREAD

**Discussion by pct-cclausen** [keycloak/keycloak#30462] [Q&A]
I am trying to upgrade to keycloak 25, however in one step of our pipeline we are trying to export the realm with all users.

We run this command:

kc.sh export --realm portal --users realm_file --file=/realm/realm.json

Keycloak 24.0.2 was happy with that.
Keycloak 25.0.0 now fails with the message:

Disabled option: '--users'
Try 'kc.sh export --help' for more information on the available options.
Specify '--help-all' to obtain information on all options and their availability.

The help it suggests lists the option in the exact same way it is used here. I could not find any information on changes in this area in the migration guide or the keycloak documentation.

**Comment by rhafer** [comment]
I ran into the same issue. A workaround seems to be to use the `--dir` option and export into a directory instead of exporting into a specific file. Seems that `--users` only work with `--dir` currently.

**Comment by SebastEnn** [comment]
For the people finding this and want an example of the workaround:

`kc.sh export --realm your_realm --users realm_file --dir=/your/export/directory`

The realm will be exported to /your/export/directory/your_realm-realm.json

According to [this reply](https://github.com/keycloak/keycloak/issues/31515#issuecomment-2245168581), the issue won't be resolved in the near future.
